### DISCUSSION TITLE
Capturing a single image

### DISCUSSION THREAD

**Discussion by alexandrefresnais** [videokit-ai/videokit#50] [Q&A] [answered]
Hi!

I was wondering if there was an easy way to capture a single image with VideoKit. I would like my recorder to support both video recording and image recording.
I have seen the JPEG recorder but it feels a bit hacky to force it to take a single frame.

Is there any way to achieve this ?

Thank you very much.

Regards,

**Comment by olokobayusuf** [comment, marked as answer]
Hey @alexandrefresnais, for capturing screenshots I'd recommend using Unity's [`ScreenCapture`](https://docs.unity3d.com/ScriptReference/ScreenCapture.CaptureScreenshot.html) API. You can use a JPEG `MediaRecorder` from VideoKit, but it's intended for image sequences, not single images.

**Reply by alexandrefresnais** [reply]
Hey, thank you very much for your answer. The thing is that `ScreenCapture` does not allow me to capture a single camera, thus captures UI. For sure this can be achieved with other ways, like I do now with RenderTextures.
I wanted to make sure I did not miss a way to merge my camera capture system with the `VideoRecorder`, as I saw the `ImageAsset`component.

Thanks again for answering me!

**Reply by olokobayusuf** [reply]
Yup I think the best way would be to use a JPEG `MediaRecorder` from code. You might be able to hack the `VideoKitRecorder` to take a screenshot for you, but I have no clue whether it will actually work:
```csharp
VideoKitRecorder recorder = ...;
// Start recording to a JPEG sequence
recorder.format = MediaFormat.JPEG;
recorder.OnRecordingCompleted.AddListener(OnRecord);
await recorder.StartRecording();
// Wait for a frame // Might need to wait for two frames instead, test this
await Task.Yield();
// Stop recording
await recorder.StopRecording();

void OnRecord (MediaAsset asset) {
    var imageSequence = asset as MediaSequenceAsset;
    var image = imageSequence.assets[0] as ImageAsset;
    // Do stuff with image...
}
```

**Reply by alexandrefresnais** [reply]
Thank you very much, will surely give it a try !
